xarray/tests/test_dataset.py

```diff
@@ -1748,6 +1748,23 @@ def test_reindex(self):
         actual = ds.reindex(x=[0, 1, 3], y=[0, 1])
         assert_identical(expected, actual)
 
+    @requires_sparse
+    def test_reindex_sparse(self):
+        data = create_test_data()
+        dim3 = list("abdeghijk")
+        actual = data.reindex(dim3=dim3, sparse=True)
+        expected = data.reindex(dim3=dim3, sparse=False)
+        for k, v in data.data_vars.items():
+            np.testing.assert_equal(actual[k].data.todense(), expected[k].data)
```

Comment:
Currently, `assert_equal` cannot be used as we need to explicitly densify the array for the comparison.